app/src/main/java/com/example/pygmyhippo/user/EventFragment.java

```diff
@@ -63,24 +80,108 @@ public Event hardcodeEvent() {
+
+        // Container Layouts
+        adminConstraint = binding.aActionsConstraint;
+
+        handler = new EventDB();
+
+        signedInAccount = EventFragmentArgs.fromBundle(getArguments()).getSignedInAccount();
+        setPermissions();
+
+        String navigationEventID = EventFragmentArgs.fromBundle(getArguments()).getEventID();
+        getEvent(navigationEventID);
+
+        // TODO: add actual database stuff here, where user is added into the events list
+        // for now, this is hardcoding to figure out structure of entrant
+        entrant = new Entrant(
+                "123",
+                Entrant.EntrantStatus.invited
+        );
 
         FloatingActionButton backButton = view.findViewById(R.id.u_backButtonToQRView);
         backButton.setOnClickListener(view1 -> {
             Navigation.findNavController(view1).navigate(R.id.action_eventFragment_to_scanQRcodeFragment);
         });
 
-        TextView eventNameView = view.findViewById(R.id.u_eventNameView);
-        TextView eventDateView = view.findViewById(R.id.u_eventDateView);
-        TextView eventTimeView = view.findViewById(R.id.u_eventTimeView);
-        TextView eventOrganizerView = view.findViewById(R.id.u_organizerNameView);
-        TextView eventLocationView = view.findViewById(R.id.u_eventLocationView);
-        TextView eventCostView = view.findViewById(R.id.u_eventCostView);
-        TextView eventAboutDescriptionView = view.findViewById(R.id.u_aboutEventDescriptionView);
+        registerButton.setOnClickListener(buttonView -> {
+            registerUser();
+        });
+
+        deleteEventButton.setOnClickListener(buttonView -> {
+            handler.deleteEventByID(event.getEventID(), this);
+        });
+
+        deleteQRCodeButton.setOnClickListener(buttonView -> {
+            Log.d("EventFragment", "Delete QR Code Button pressed");
+            // TODO: Delete QR Code.
+        });
 
-        // set with hardcoded values, set the view in the same wy form the event received from the constructor
-        // TODO: database event info to be done here
-        event = hardcodeEvent();
+        return view;
+    }
+
+    @Override
+    public void OnComplete(@NonNull ArrayList<Event> docs, int queryID, int flags) {
+        if (queryID == 0) {
+            if (flags == DBOnCompleteFlags.SINGLE_DOCUMENT.value) {
+                event = docs.get(0);
+                populateTextFields();
+            }
+        } else if (queryID == 1) {
+            if (flags == DBOnCompleteFlags.SUCCESS.value) {
+                Log.d("EventFragment", "Successfully deleted event. Navigating back to all events");
+                Bundle navArgs = new Bundle();
+                navArgs.putParcelable("signedInAccount", signedInAccount);
+                navController.navigate(R.id.admin_navigation_all_events, navArgs);
+            } else {
+                Log.d("EventFragment", "Error in deleting event.");
+            }
+        }
+    }
+
```

Comment:
Just wanna stay up to speed here, does this populate the event fragment texts with an event from the database?